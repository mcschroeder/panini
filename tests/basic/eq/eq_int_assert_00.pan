eq : (a:ℤ) → (b:ℤ) → { c:𝔹 | c = true ⟺ a = b }
assert : { b:𝔹 | b = true } → 𝟙

f00 = 
  let p = eq 1 1 in 
  assert p

f01 : unit
f01 = 
  let p = eq 1 1 in 
  assert p

f10 = 
  let x = 1 in
  let p = eq x 1 in 
  assert p

f11 : unit
f11 = 
  let x = 1 in
  let p = eq x 1 in 
  assert p
