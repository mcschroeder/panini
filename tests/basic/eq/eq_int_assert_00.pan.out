f00 : 𝟙
f01 : 𝟙
f10 : 𝟙
f11 : 𝟙